pyproject.toml

```diff
@@ -31,7 +33,7 @@ classifiers = [
     "Development Status :: 4 - Beta",
     "Typing :: Typed",
 ]
-[project.optional-dependencies]
+[dependency-groups]
```

Comment:
now that we have dependency groups, isn't it time to define and use separate `unit`, `lint`, `integration` groups that we can use for the various tox envs instead of always installing the whole dev group?

Reply:
I'm agnostic to this, but I think its an orthogonal question.  Groups existed in `optional-dependencies` too, but for (idk what reasons) we still put everything in `dev`.  imo this is a separate conversation, but I'm open if you disagree